interfaces.go

```diff
@@ -168,15 +168,29 @@ type Mutex interface {
 // Logger implements logger abstraction
 type Logger interface {
 	Debug(args ...interface{})
+	DebugWithError(msg string, err error)
 	Debugf(format string, args ...interface{})
+	Debugb() EventBuilder
```

Comment:
Why new methods with `b` return `EventBuilder`, but methods `WithError` are not?

Reply:
Because `WithError` is a convenient shorthand to write logs with error messages in one line.

Also `Msg` should be called when all the tags are set, and `WithError` calls `Msg` , thus no more tags can be set

Reply:
`tag` is a common name for all key-value pairs set for a sertain event.

`WithError` methods were removed after duscussion with @kissken.

Added comment to Msg in EventBuilder interface.